R/watcher.r

```diff
@@ -47,11 +47,10 @@ watch <- function(path, callback, pattern = NULL, hash = TRUE) {
 #' @return a digest of the file, or NA if it doesn't exist.
 #' @keywords internal
 safe_digest <- function(path) {
-  if (!file.exists(path)) return(NA_character_)
-  if (is_directory(path)) return(NA_character_)
-  if (!is_readable(path)) return(NA_character_)
-
-  digest::digest(path, file = TRUE)
+  tryCatch(
+    digest::digest(path, file = TRUE),
+    error = function(e) NA_character_
+  )
```

Comment:
Thanks for the feedback. Unfortunately, the previous approach isn't perfectly robust -- someone could modify `path` between the `is_readable()` and `digest()` calls. I agree that this is unlikely, still it looks like only the `tryCatch` approach is really error-prone. That's why I pinged you -- if `digest()` returned `NA` everything would be much easier here.